### ISSUE TITLE
no-unsafe-any: don't allow implicit string coercion

### ISSUE THREAD

**Issue by ajafff** [palantir/tslint#3441]
Symbols cannot be implicitly converted to string. Instead an error is thrown:

```ts
var symbol = Symbol('foo');

console.log(String(symbol)); // ok
console.log(`${symbol}`); // Runtime Error
console.log("" + symbol); // Runtime Error
```

TypeScript currently doesn't check this special case. But when they do, we should also make `no-unsafe-any` stricter.
Ref: https://github.com/Microsoft/TypeScript/issues/19666

**Comment by JoshuaKGoldberg** [comment]
This was separately added as `strict-string-expressions`. ✔️
